src/types/index.ts

```diff
@@ -341,38 +346,45 @@ export interface NullableStrError {
   error: string | null;
 }
 
-export interface TextFileWatcherReadyArgs {
+export type TextChange = {
+  from: number;
+  to?: number;
+  insert?: string;
+};
```

Comment:
We can allow `ChangeSet` in the future, but intentionally leaving codemirror out as an implementation detail for now @Conner1115